contracts/settling_game/modules/food/library.cairo

```diff
@@ -0,0 +1,194 @@
+# -----------------------------------
+# ____Food Library
+#   Food
+#
+# MIT License
+# -----------------------------------
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.starknet.common.syscalls import get_block_timestamp
+from starkware.cairo.common.uint256 import Uint256, uint256_eq
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.cairo.common.math import unsigned_div_rem, assert_not_zero, assert_le, assert_nn
+from starkware.cairo.common.math_cmp import is_le
+
+from contracts.settling_game.utils.constants import (
+    FARM_LENGTH,
+    MAX_HARVESTS,
+    HARVEST_LENGTH,
+    STORE_HOUSE_SIZE,
+)
+from contracts.settling_game.utils.game_structs import (
+    RealmData,
+    RealmBuildingsIds,
+    HarvestType,
+    FoodBuildings,
+)
+from contracts.settling_game.utils.constants import SHIFT_41
+from contracts.settling_game.utils.general import unpack_data
+
+namespace Food:
+    # @notice Calculates how many available farms to harvest
+    # max of 3 full harvests accure
+    # if more farms than MAX, return MAX and decayed farms. How will loose these harvests.
+    func calculate_harvest{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr : BitwiseBuiltin*,
+    }(update_time : felt, block_timestamp : felt) -> (
+        total_farms : felt, total_remaining : felt, decayed_farms : felt
+    ):
+        alloc_locals
+
+        let time_since_update = block_timestamp - update_time
+
+        # TODO add max days can accure
+        let (total_farms, remainding_crops) = unsigned_div_rem(time_since_update, HARVEST_LENGTH)
+
+        let (le_max_farms) = is_le(total_farms, MAX_HARVESTS + 1)
+
+        if le_max_farms == TRUE:
+            return (total_farms, remainding_crops, 0)
```

Comment:
Is this returning of 0 for decayed farms here correct? Wouldn't that mean there's no point in building more food buildings beyond 7 (`BASE_HARVESTS / 4 + 1`) if I don't need the food and they would never decay?

Reply:
Farms decay if you have not harvested them. So if you harvest less than the MAX_HARVESTS you don't have any decayed farms.